plugins/processors/converter/converter_test.go

```diff
@@ -451,41 +402,123 @@ func TestConverter(t *testing.T) {
 					Integer: []string{"int_*"},
 				},
 			},
-			input: Metric(
-				metric.New(
+			input: testutil.MustMetric(
+				"cpu",
+				map[string]string{},
+				map[string]interface{}{
+					"int_a":   "1",
+					"int_b":   "2",
+					"float_a": 1.0,
+				},
+				time.Unix(0, 0),
+			),
+			expected: []telegraf.Metric{
+				testutil.MustMetric(
 					"cpu",
 					map[string]string{},
 					map[string]interface{}{
-						"int_a":   "1",
-						"int_b":   "2",
+						"int_a":   int64(1),
+						"int_b":   int64(2),
 						"float_a": 1.0,
 					},
 					time.Unix(0, 0),
 				),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.converter.Log = testutil.Logger{}
+
+			err := tt.converter.Init()
+			require.NoError(t, err)
+			actual := tt.converter.Apply(tt.input)
+
+			testutil.RequireMetricsEqual(t, tt.expected, actual)
+		})
+	}
+}
+
+func TestMeasurement(t *testing.T) {
+	tests := []struct {
+		name      string
+		converter *Converter
+		input     telegraf.Metric
+		expected  []telegraf.Metric
+	}{
+		{
+			name: "measurement from tag",
+			converter: &Converter{
+				Tags: &Conversion{
+					Measurement: []string{"filepath"},
+				},
+			},
+			input: testutil.MustMetric(
+				"file",
+				map[string]string{
+					"filepath": "/var/log/syslog",
+				},
+				map[string]interface{}{
+					"msg": "howdy",
+				},
+				time.Unix(0, 0),
 			),
-			expected: Metric(
-				metric.New(
-					"cpu",
+			expected: []telegraf.Metric{
+				testutil.MustMetric(
+					"/var/log/syslog",
```

Comment:
#6970 talks about using filenames as the measurement name.  I don't think they'll want to use the full path.  Is there a way to get just the basename of the file input plugin's file_tag?

Reply:
You would need to use another processor to modify the tag value first, it ought to be possible to do this with the regex processor though.  We could potentially have a `filepath` processor that does common transformations, or perhaps it would fit in with the `strings` processor.